File: data_structures/programmers_add_not_exist_numbers.c
/*
테스트 1 〉	통과 (0.01ms, 4.09MB)
테스트 2 〉	통과 (0.01ms, 4.18MB)
테스트 3 〉	통과 (0.01ms, 4.36MB)
테스트 4 〉	통과 (0.01ms, 4.16MB)
테스트 5 〉	통과 (0.01ms, 4.16MB)
테스트 6 〉	통과 (0.01ms, 4.16MB)
테스트 7 〉	통과 (0.01ms, 4.16MB)
테스트 8 〉	통과 (0.01ms, 4.1MB)
테스트 9 〉	통과 (0.01ms, 4.09MB)
*/

#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// numbers_len은 배열 numbers의 길이입니다.
int solution(int numbers[], size_t numbers_len) {
    int answer = 0;
    
    for (int i = 0; i <= 9; i++) answer += i;
    for (int i = 0; i < numbers_len; i++) answer -= numbers[i];
    
    return answer;
}

File: data_structures/programmers_add_not_exist_numbers.java
/*
테스트 1 〉	통과 (0.01ms, 73.6MB)
테스트 2 〉	통과 (0.01ms, 74.3MB)
테스트 3 〉	통과 (0.02ms, 77.4MB)
테스트 4 〉	통과 (0.02ms, 73.1MB)
테스트 5 〉	통과 (0.02ms, 78.3MB)
테스트 6 〉	통과 (0.01ms, 73.6MB)
테스트 7 〉	통과 (0.03ms, 75.3MB)
테스트 8 〉	통과 (0.02ms, 76.4MB)
테스트 9 〉	통과 (0.01ms, 73.9MB)
*/

class Solution {
    public int solution(int[] numbers) {
        int answer = 0;
        
        for (int i = 0; i <= 9; i++)
            answer += i;
        
        for (int n : numbers)
            answer -= n;
        
        return answer;
    }
}
